src/uidl-definitions/schemas/component.json

```diff
@@ -200,7 +204,7 @@
           "additionalProperties": false,
           "required": ["type", "defaultValue"],
           "properties": {
-            "type": {"type": "string"},
+            "type": {"type":"string", "enum": ["string", "boolean", "number", "object"]},
```

Comment:
same thing is needed for the type inside `propDefinition`
and you should also include `array` and `function`